provider/balance_checker_test.go

```diff
@@ -49,7 +49,7 @@ func balanceCheckerForTest(t *testing.T, balance int64) (*scaffold, *balanceChec
 
```

Comment:
@troian I am still getting some races reported around this but only in the test mock code, so I have refactored the test now.

Reply:
👍